### PR TITLE
refactor(ui): refactor capitalization of platform name and sub types

### PR DESCRIPTION
**Pull request by Masterchen09**
I was really surprised and actually can't believe that no one else has noticed it yet and asked about it on Slack, but the capitalization of the platform names (and the sub types) is currently inconsistent: When you navigate through the entities, lineage, etc. sometimes the platform names are correctly capitalized and sometimes not. Therefore, I have took the time and checked all occurrences of capitalizeFirstLetter and capitalizeFirstLetterOnly in relation to the platform name and the sub types:

- If a display name is available for a platform, the display name should be visible.
- The capitalization of the display name of a platform should never be changed, because we should assume that the display name is already correct.
- If no display name is available the name of the platform should be visible.
- The name of the platform seems to have been capitalized in the past, therefore it should be capitalized everywhere where it is used when there is no display name available.
    - Even though we could discuss whether this makes sense or not: Assuming that we do not have a display name for the MySQL platform, Mysql after capitalization instead of mysql is not really better, as the correct display name would be MySQL? Therefore, we could also think about to completely remove the capitalization for the platform names?
- For the sub types it is a bit easier, as there is currently no display name, therefore everywhere where the sub types are displayed, they should be capitalized.
    - This is also something we could discuss: Does it makes sense to change the capitalization of the sub types, I have found places where the sub types are ingested with the correct casing (see [here](https://github.com/datahub-project/datahub/blob/87b3a5d0fcb508e9abb92322bc94a4ce47329e08/metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py#L708)), but there are also places where the sub types are ingested with lowercase letters only (see [here](https://github.com/datahub-project/datahub/blob/7607c04ffa790fec9e96e09a9a8b10534818572d/metadata-ingestion/src/datahub/ingestion/source/kafka.py#L294)). I am not sure whether there are some best practices how sub types should be ingested (or if there are any ideas/plans for enhancing this in general, e.g. by introducing a new sub type entity with a display name where the sub types aspect is then only referencing to the sub types entity), but I think as long lowercase-only sub types are valid, the sub types should be capitalized in the UI.
- In all cases capitalizeFirstLetterOnly and not capitalizeFirstLetter should be used to support sub types consisting of multiple words, where the casing is already correct (e.g., "My Super Dashboard" instead of "My super dashboard" after using capitalizeFirstLetter):
    - Alternatively, we could also introduce a new capitalization function which only changes the capitalization when all letters of the sub type are in lowercase...but capitalizeFirstLetterOnly is also sufficient for most of the cases (capitalizeFirstLetterOnly does only not work for cases where the first character should be in lowercase intentionally).

P.S.: The removal of the meta-favicon.ico has nothing to do with the capitalization, but I think it was added by mistake, because it is not used anywhere, therefore I have removed it.

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/chart/ChartEntity.tsx [hunk 1]
```diff
@@ -197,9 +198,9 @@ export class ChartEntity implements Entity<Chart> {
     getLineageVizConfig = (entity: Chart) => {
         return {
             urn: entity.urn,
-            name: entity.properties?.name || '',
+            name: entity.properties?.name || entity.urn,
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/chart/ChartEntity.tsx:201]: Thank you!

File: datahub-web-react/src/app/entity/chart/preview/ChartPreview.tsx [hunk 2]
```diff
@@ -62,7 +61,6 @@ export const ChartPreview = ({
     snippet?: React.ReactNode | null;
 }): JSX.Element => {
     const entityRegistry = useEntityRegistry();
-    const capitalizedPlatform = capitalizeFirstLetter(platform);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/chart/preview/ChartPreview.tsx:65]: Thank you

File: datahub-web-react/src/app/entity/container/preview/Preview.tsx [hunk 3]
```diff
@@ -62,7 +63,7 @@ export const Preview = ({
     externalUrl?: string | null;
 }): JSX.Element => {
     const entityRegistry = useEntityRegistry();
-    const typeName = (subTypes?.typeNames?.length && subTypes?.typeNames[0]) || 'Container';
+    const typeName = capitalizeFirstLetterOnly(subTypes?.typeNames?.[0]) || 'Container';
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/container/preview/Preview.tsx:66]: Minor: Will this work when 

subTypes?.typeNames is an empty array?. If not, seems a bit worrisome to remove the first check.

File: datahub-web-react/src/app/entity/mlFeature/MLFeatureEntity.tsx [hunk 4]
```diff
@@ -169,7 +169,7 @@ export class MLFeatureEntity implements Entity<MlFeature> {
             // eslint-disable-next-line
             icon: entity?.['featureTables']?.relationships?.[0]?.entity?.platform?.properties?.logoUrl || undefined,
             // eslint-disable-next-line
-            platform: entity?.['featureTables']?.relationships?.[0]?.entity?.platform?.name,
+            platform: entity?.['featureTables']?.relationships?.[0]?.entity?.platform,
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/mlFeature/MLFeatureEntity.tsx:172]: Are we confident in this change?

File: datahub-web-react/src/app/entity/shared/containers/profile/header/EntityHeader.tsx [hunk 5]
```diff
@@ -89,15 +89,13 @@ export const EntityHeader = ({
...
-    const platformName = capitalizeFirstLetterOnly(basePlatformName);
+    const platformName = getPlatformName(entityData);
     const externalUrl = entityData?.externalUrl || undefined;
     const entityCount = entityData?.entityCount;
     const isCompact = React.useContext(CompactContext);
 
     const entityName = entityData?.name;
-    const subType =
-        (entityData?.subTypes?.typeNames?.length || 0) > 0 ? entityData?.subTypes?.typeNames![0] : undefined;
+    const subType = capitalizeFirstLetterOnly(entityData?.subTypes?.typeNames?.[0]) || undefined;
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/header/EntityHeader.tsx:98]: Is it safe to remove this check? entityData?.subTypes?.typeNames?.length

File: datahub-web-react/src/app/entity/shared/containers/profile/header/PlatformContent/PlatformContentContainer.tsx [hunk 6]
```diff
@@ -15,19 +15,18 @@ export function getDisplayedEntityType(
     entityRegistry: EntityRegistry,
     entityType: EntityType,
 ) {
-    const entityTypeCased =
-        (entityData?.subTypes?.typeNames?.length && capitalizeFirstLetterOnly(entityData?.subTypes.typeNames[0])) ||
-        entityRegistry.getEntityName(entityType);
-    return entityData?.entityTypeOverride || entityTypeCased || '';
+    return (
+        entityData?.entityTypeOverride ||
+        capitalizeFirstLetterOnly(entityData?.subTypes?.typeNames?.[0]) ||
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/header/PlatformContent/PlatformContentContainer.tsx:20]: is it safe to remove the check entityData?.subTypes?.typeNames?.length

What if type names does not exist?

Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/containers/profile/header/PlatformContent/PlatformContentContainer.tsx:20]: (I see that the method can deal with undefined), but I think the other case is when the array exists, but is empty (technically a possible state)

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Validations/DatasetAssertionsList.tsx [hunk 7]
```diff
@@ -125,7 +130,12 @@ export const DatasetAssertionsList = ({ assertions, onDelete }: Props) => {
                                     width={20}
                                     src={record.platform.properties?.logoUrl}
                                 />
-                            )) || <Typography.Text>{record.platform.properties?.displayName}</Typography.Text>}
+                            )) || (
+                                <Typography.Text>
+                                    {record.platform.properties?.displayName ||
+                                        capitalizeFirstLetterOnly(record.platform.name)}
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Validations/DatasetAssertionsList.tsx:136]: Thank you for adding this!

File: datahub-web-react/src/app/entity/shared/utils.ts [hunk 8]
```diff
@@ -71,7 +72,7 @@ export const singularizeCollectionName = (collectionName: string): string => {
 };
 
 export function getPlatformName(entityData: GenericEntityProperties | null) {
-    return entityData?.platform?.properties?.displayName || entityData?.platform?.name;
+    return entityData?.platform?.properties?.displayName || capitalizeFirstLetterOnly(entityData?.platform?.name);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/utils.ts:75]: Thank you!!

File: datahub-web-react/src/app/ingest/source/IngestionSourceList.tsx [hunk 9]
```diff
@@ -266,7 +266,7 @@ export const IngestionSourceList = () => {
                 message.success({ content: 'Removed ingestion source.', duration: 2 });
                 const newRemovedUrns = [...removedUrns, urn];
                 setRemovedUrns(newRemovedUrns);
-                setTimeout(function () {
+                setTimeout(() => {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/ingest/source/IngestionSourceList.tsx:269]: Not sure why these changes are strictly necessary, but we will leave them

File: datahub-web-react/src/app/lineage/LineageEntityNode.tsx [hunk 10]
```diff
@@ -109,12 +109,11 @@ export default function LineageEntityNode({
...
 
-    let platformDisplayText = capitalizeFirstLetter(
-        node.data.platform?.properties?.displayName || node.data.platform?.name,
-    );
+    let platformDisplayText =
+        node.data.platform?.properties?.displayName || capitalizeFirstLetterOnly(node.data.platform?.name);
     if (node.data.siblingPlatforms && !isHideSiblingMode) {
         platformDisplayText = node.data.siblingPlatforms
-            .map((platform) => platform.properties?.displayName || platform.name)
+            .map((platform) => platform.properties?.displayName || capitalizeFirstLetterOnly(platform.name))
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/lineage/LineageEntityNode.tsx:116]: Thank you!

File: datahub-web-react/src/app/lineage/__tests__/LineageEntityView.test.tsx [hunk 11]
```diff
@@ -32,7 +32,7 @@ describe('LineageEntityView', () => {
     it('should not render a divider if there is no platform name', () => {
         const datasetNoPlatformName = {
             ...dataset1,
-            platform: { ...dataset1.platform, properties: { displayName: '' } },
+            platform: { ...dataset1.platform, name: '', properties: { displayName: '' } },
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/lineage/__tests__/LineageEntityView.test.tsx:35]: Why empty string ?

File: datahub-web-react/src/app/recommendations/renderer/component/EntityNameList.tsx [hunk 12]
```diff
@@ -118,14 +119,12 @@ export const EntityNameList = ({
                 const additionalProperties = additionalPropertiesList?.[index];
                 const genericProps = entityRegistry.getGenericEntityProperties(entity.type, entity);
                 const platformLogoUrl = genericProps?.platform?.properties?.logoUrl;
-                const platformName =
-                    genericProps?.platform?.properties?.displayName ||
-                    capitalizeFirstLetter(genericProps?.platform?.name);
+                const platformName = getPlatformName(genericProps);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/recommendations/renderer/component/EntityNameList.tsx:122]: Amazing!

**Review by jjoyce0510 — COMMENTED** [review]
So firstly, I in large part agree with your assessment.

1. Platform Name should only be capitalized (first letter) if there is not display name set.
2. Sub Type name should only be capitalized for the first letter (in all cases)

It seems that this PR covers it!

I think there's just one case that was sometimes previously covered but now has been removed in some places: checking for the presence of the subTypes array itself. 

Other than this, I think this PR is largely ready go. I am very grateful you took the time to do this!

File: datahub-web-react/src/app/ingest/source/IngestionSourceList.tsx (hunk 9 above)
Comment (Masterchen09) [reply datahub-web-react/src/app/ingest/source/IngestionSourceList.tsx:269]: Oh, I forgot to mention it: ESLint reported it as an unexpected unnamed function...everywhere else in the code an arrow function was used, so I changed it.

File: datahub-web-react/src/app/lineage/__tests__/LineageEntityView.test.tsx (hunk 11 above)
Comment (Masterchen09) [reply datahub-web-react/src/app/lineage/__tests__/LineageEntityView.test.tsx:35]: Before the changes in LinageEntityView.tsx only the displayName of the platform was used, therefore the name of the platform must also be set to an empty string (or undefined) for this test to be successful. If I think about it again, in theory we could also remove this test completely, because the platform name should never be empty, however as the platform in GenericEntityProperties is optional it is possible that getPlatformName returns undefined...in this case the test has somehow the right to exist...

**Review by jjoyce0510 — APPROVED** [review]
LGTM - Thank you for adding!

**Comment by jjoyce0510** [comment]
One more thing! Looks like this has caused a smoke test to fail.. Can you change `siblings.js ` to lower case "Dbt" to "dbt" here: https://github.com/datahub-project/datahub/blob/master/smoke-test/tests/cypress/cypress/integration/siblings/siblings.js#L127

**Comment by jjoyce0510** [comment]
Once we have that small change, we are good to merge.

Thanks
John
